packages/react-native-reanimated/src/common/style/processors/filter.ts

```diff
@@ -0,0 +1,198 @@
+
+import { ReanimatedError } from '../../errors';
+import type {
+  FilterArray,
+  ParsedDropShadow,
+  ValueProcessor,
+} from '../../types';
+import { isLength } from '../../utils/guards';
+import { processColor } from './colors';
+
+// Capture filter functions and their content eg "brightness(0.5) opacity(1)" => [["brightness(0.5)", "brightness", "0.5"], ["opacity(1)", "opacity", "1"]]
+const FILTER_REGEX = /([\w-]+)\(([^()]*|\([^()]*\)|[^()]*\([^()]*\)[^()]*)\)/g;
+// Capture two groups: current transform value and optional unit -> "21.37px" => ["21.37px", "21.37", "px"]
+const FILTER_VALUE_REGEX = /^([-+]?\d*\.?\d+)([a-z%]*)$/;
+// Capture drop-shadow parts "10px 5px 5px #888888" => ["10px", "5px", "5px", "#888888"]
+const DROP_SHADOW_REGEX = /[^,\s()]+(?:\([^()]*\))?/g;
+
+export const ERROR_MESSAGES = {
+  invalidFilter: (filter: string) => `Invalid filter property: ${filter}`,
+  invalidColor: (color: string, dropShadow: string) =>
+    `Invalid color "${color}" in dropShadow "${dropShadow}".`,
+};
+
+type SingleFilterValue = {
+  number: number;
+  unit: string;
+};
+
+const parseHueRotate = (value: SingleFilterValue): number | undefined => {
+  const { number, unit } = value;
+  if (number === 0) {
+    return 0;
+  }
+  if (unit !== 'deg' && unit !== 'rad') {
+    throw new ReanimatedError(
+      ERROR_MESSAGES.invalidFilter(`hueRotate(${number}${unit})`)
+    );
+  }
+  return unit === 'rad' ? (180 * number) / Math.PI : number;
+};
+
+const parseBlur = (value: SingleFilterValue): number | undefined => {
+  const { number, unit } = value;
+  if ((unit && unit !== 'px') || number < 0) {
+    return undefined;
+  }
+  return number;
+};
+
+const LENGTH_MAPPINGS = ['offsetX', 'offsetY', 'standardDeviation'] as const;
+
+const parseDropShadowString = (value: string) => {
+  const match = value.match(DROP_SHADOW_REGEX) ?? [];
+  const result: DropShadowValue = { offsetX: 0, offsetY: 0 };
+  let foundLengthsCount = 0;
+
+  match.forEach((part) => {
+    if (isLength(part) || !!parseFloat(part)) {
+      const dropShadowValue = part.trim().match(FILTER_VALUE_REGEX);
+      if (!dropShadowValue) {
+        throw new ReanimatedError(
+          ERROR_MESSAGES.invalidFilter(`dropShadow(${part})`)
+        );
+      }
+      result[LENGTH_MAPPINGS[foundLengthsCount++]] = dropShadowValue[1];
+    } else {
+      result.color = part.trim();
+    }
+  });
+
+  return result;
+};
+const parseDropShadow = (value: string | DropShadowValue): ParsedDropShadow => {
+  const dropShadow =
+    typeof value === 'string' ? parseDropShadowString(value) : value;
+  const {
+    color = '#000',
+    offsetX = 0,
+    offsetY = 0,
+    standardDeviation = 0,
+    ...rest
+  } = dropShadow;
+
+  const processedColor = processColor(color);
+
+  if (processedColor === undefined || processedColor === null) {
+    throw new ReanimatedError(
+      ERROR_MESSAGES.invalidColor(color as string, JSON.stringify(dropShadow))
+    );
+  }
```

Comment:
```suggestion
  const processedColor = processColor(color);
```

`processColor` throws an error itself so you shouldn't throw error here in this way. If you prefer to add a meaningful error saying that the color in the `boxShadow` filter is invalid, you'd have to use the `processColorInitially` function or extract the part of the `processColor` that doesn't throw an error and use it here.

Reply:
was using it because of it's usage in `shadows.ts` so good to know we need to change it in two places :P